internal/plugin/server.go

```diff
@@ -326,69 +326,71 @@ func (plugin *NvidiaDevicePlugin) Allocate(ctx context.Context, reqs *pluginapi.
 func (plugin *NvidiaDevicePlugin) getAllocateResponse(requestIds []string) (*pluginapi.ContainerAllocateResponse, error) {
 	deviceIDs := plugin.deviceIDsFromAnnotatedDeviceIDs(requestIds)
 
-	responseID := uuid.New().String()
-	response, err := plugin.getAllocateResponseForCDI(responseID, deviceIDs)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get allocate response for CDI: %v", err)
+	// Create an empty response that will be updated as required below.
+	response := &pluginapi.ContainerAllocateResponse{
+		// Envs is set to the empty map to support insertions.
+		Envs: make(map[string]string),
+	}
+
+	if plugin.cdiEnabled {
+		responseID := uuid.New().String()
+		if err := plugin.updateAllocateResponseForCDI(response, responseID, deviceIDs...); err != nil {
+			return nil, fmt.Errorf("failed to get allocate response for CDI: %v", err)
+		}
+		return response, nil
 	}
 
+	if plugin.config.Sharing.SharingStrategy() == spec.SharingStrategyMPS {
+		plugin.updateAllocatResponseForMPS(response, deviceIDs...)
+	}
 	if plugin.deviceListStrategies.Includes(spec.DeviceListStrategyEnvvar) {
-		response.Envs = plugin.apiEnvs(plugin.deviceListEnvvar, deviceIDs)
+		plugin.updateDeviceListEnvvar(response, deviceIDs...)
```

Comment:
Should this function be called `plugin.updateAllocateResponseDeviceListEnvar()` for symmetry with the names of the other calls. Or alternatively remove the `AllocateResponse` part of the other calls?

Reply:
OK. I will take another look at them.

Reply:
renamed all to `updateResponseFor*` as a balance between length and readability.